### ISSUE TITLE
Updating track "year" field doesn't update album

### ISSUE THREAD

**Issue by apocalyptech** [apocalyptech/exordium#1]
I've been re-ripping some old CDs, and some of my old rips never had the year field populated.  The new rips do, and the tracks themselves get the year updated properly, but the album year field has been staying at `0`.

Will have to decide exactly how I'd want that update to happen.  Take a look at all tracks on an album and use the one that's used on a majority of the files?  It'd be weird if tracks within an album have different years, though I suppose that's pretty likely for "non-album" tracks.

**Comment by apocalyptech** [comment]
Looks like the same issue with updating the album title itself, come to that -- recently tagged an album which had "smart" apostrophes in the album title, and I didn't notice until after I'd imported.  I really hate those, so I switched the tagging to use regular ASCII apostrophes, and the tracks themselves updated properly, as you'd hope, but the album title itself remained the same.

**Comment by apocalyptech** [comment]
Heh, I clearly jumped right on "*this* one, didn't I?

Anyway, some other update weirdness: I've been re-ripping some old albums again, and had run into this issue with the years.  But I'd also noticed that sometimes the filesizes/bitrates don't update, when the filename stays the same?

To reproduce:
1. Re-encode an album, making a filename change to *one* of the tracks
2. Do a full update -- observe that *only* the renamed file has the new values (as expected, since it's a new row0
3. `touch` all the files in the album and re-update
4. Now *all* of them have updated.

Really weird -- why doesn't that update the first time?  The files are new; they've been re-encoded.

**Comment by apocalyptech** [comment]
Some debug output from an update which triggered this, btw, since I finally remembered to check that box:

    Starting process...
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/01-a_god_in_an_alcove.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/02-telegram_sam.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/03-double_dare.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/04-the_spy_in_the_cab.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/05-in_the_flat_field.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/07-in_fear_of_fear.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/08-poison_pen.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/09-party_of_the_first_part.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/10-departure.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/11-the_three_shadows_part_2.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/12-silent_hedges.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/13-swing_the_heartache.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/14-third_uncle.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/15-ziggy_stardust.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/16-terror_couple_kill_colonel.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/17-night_time.mp3
    Updated file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/18-shes_in_parties.mp3
    Found new file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/06-st_vitus_dance.mp3
    Deleted file: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/06-st._vitus_dance.mp3
    Initial album artist: Bauhaus
    Looking at album Swing the Heartache (The BBC Sessions), artist Bauhaus, tracks 18
    tracks to update: 0, possible: 18
    Using existing album "Bauhaus / Swing the Heartache (The BBC Sessions)" for Swing the Heartache (The BBC Sessions)
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/01-a_god_in_an_alcove.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/02-telegram_sam.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/03-double_dare.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/04-the_spy_in_the_cab.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/05-in_the_flat_field.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/07-in_fear_of_fear.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/08-poison_pen.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/09-party_of_the_first_part.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/10-departure.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/11-the_three_shadows_part_2.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/12-silent_hedges.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/13-swing_the_heartache.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/14-third_uncle.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/15-ziggy_stardust.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/16-terror_couple_kill_colonel.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/17-night_time.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/18-shes_in_parties.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/01-a_god_in_an_alcove.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/03-double_dare.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/17-night_time.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/07-in_fear_of_fear.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/14-third_uncle.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/11-the_three_shadows_part_2.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/04-the_spy_in_the_cab.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/13-swing_the_heartache.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/16-terror_couple_kill_colonel.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/15-ziggy_stardust.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/08-poison_pen.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/06-st_vitus_dance.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/05-in_the_flat_field.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/09-party_of_the_first_part.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/18-shes_in_parties.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/02-telegram_sam.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/10-departure.mp3
    Processed file changes for: Darkwave/Bauhaus/Swing_the_Heartache_(The_BBC_Sessions)/12-silent_hedges.mp3

... at the end only the totally-new file for the slightly-renamed St. Vitus Dance ended up with correct info.

**Comment by apocalyptech** [comment]
And of course I'd forgotten to check the "debug" checkbox when re-running the update after `touch`ing the files again, but here's that post-`touch` output from another re-ripped album which exhibited the same behavior:

    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/01-Shes_In_Parties.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/02-Antonin_Artaud.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/03-Wasp.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/04-King_Volcano.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/05-Who_Killed_Mr_Moonlight.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/06-Slice_of_Life.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/07-Honeymoon_Croon.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/08-Kingdoms_Coming.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/09-Burning_From_The_Inside.mp3
    Updated file: Darkwave/Bauhaus/Burning_From_The_Inside/10-Hope.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/01-Shes_In_Parties.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/02-Antonin_Artaud.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/03-Wasp.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/04-King_Volcano.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/05-Who_Killed_Mr_Moonlight.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/06-Slice_of_Life.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/07-Honeymoon_Croon.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/08-Kingdoms_Coming.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/09-Burning_From_The_Inside.mp3
    Processed file changes for: Darkwave/Bauhaus/Burning_From_The_Inside/10-Hope.mp3

... interestingly it does *not* include those `Looking at album` and `tracks to update` lines.

Also verified while doing some more than when there's no renamed tracks in the update, all tracks in the album get updated properly (which makes sense, since the `touch` triggers things off properly).

**Comment by apocalyptech** [comment]
Mmf, well, I commited a fix for the original "year" updates, and I can't seem to reproduce the other issues that I was having.  I guess I'm gonna just close this out for now but keep an eye out.  Weird.  The year fix shouldn't've really had an effect apart from updating the year properly.

**Comment by apocalyptech** [comment]
Okay, groovy, some more re-ripping activity and I managed to reproduce this.  Basically what was happening was we'd been adding all files with changed mtimes to a `to_update` list, and this was working fine, and that list ends up getting saved out to the database.  *But,* those files were getting added *again* to the same list, and the data stored in that appended version was the original.  So the database would get updated properly but then immediately overwritten with old data again.

(the fix for that is in a8958f272b158c41d940a15448ed0cee6678936f btw)

I think that the Year fix I did earlier is probably still required -- I'll verify that in just a bit.  Really these should've been separate Issues.
